### PR TITLE
Allow reg subject to be bids filters; refactor getting stanford data into temp folder

### LINKED ISSUE #447: Reg subject should take BIDS filters instead of a path when customizing.

### PR DESCRIPTION
**Pull request by 36000**
- [x] Add a test, move it to nightly if necessary

Resolves #447

### CODE REVIEW

**Comment by 36000** [comment]
generate b0 and put it in an imaginary pipeline

**Comment by 36000** [comment]
@arokem this is ready for review / merge

File: AFQ/api.py [hunk 1]
```diff
@@ -225,9 +225,9 @@ def __init__(self,
             [REGISTRATION] Maximum b value you want to use
             from the dataset (other than b0).
             If None, there is no maximum limit. Default: None
-        reg_subject : str or Nifti1Image, optional
+        reg_subject : str, Nifti1Image, dict, optional
             [REGISTRATION] The source image data to be registered.
-            Can either be a Nifti1Image, a path to a Nifti1Image, or
+            Can either be a Nifti1Image, bids filters for a Nifti1Image, or
```
[the remark this answers is not in the document]
Comment (arokem) [reply AFQ/api.py:230]: Or maybe "if" on the next line shouldn't be caps

**Review by arokem — COMMENTED** [review]
Looks good. I had a couple of small docstring comments, but you  can merge once you've fixed those.

**Pull request merged**